python/taichi/lang/expr.py

```diff
@@ -39,6 +39,12 @@ def __init__(self, *args, tb=None, dtype=None):
             self.ptr.set_tb(self.tb)
         self.ptr.type_check(impl.get_runtime().prog.config)
 
+    def __getitem__(self, *indices):
```

Comment:
Shall we add this to `impl.subscript`? Calling `__getitem__` is for non-Taichi types: https://github.com/taichi-dev/taichi/blob/2bf8cef297cd0b617eef6addcd4cb2cd1708a4dc/python/taichi/lang/impl.py#L227-L228